examples/routing_service/routing_service_file_adapter/cpp/FileStreamWriter.cxx

```diff
@@ -0,0 +1,60 @@
+/*****************************************************************************/
+/*         (c) Copyright, Real-Time Innovations, All rights reserved.        */
+/*                                                                           */
+/*         Permission to modify and use for internal purposes granted.       */
+/* This software is provided "as is", without warranty, express or implied.  */
+/*                                                                           */
+/*****************************************************************************/
+
+#include "FileStreamWriter.hpp"
+#include <rti/topic/to_string.hpp>
+#include <rti/core/Exception.hpp>
+
+using namespace rti::routing;
+using namespace rti::routing::adapter;
+using namespace rti::community::examples;
+
+const std::string FileStreamWriter::OUTPUT_FILE_PROPERTY_NAME = "example.adapter.output_file";
+
+FileStreamWriter::FileStreamWriter(const PropertySet &property)
+{
+    std::string output_file_name;
+    for (const auto &it: property) {
+        if (it.first == OUTPUT_FILE_PROPERTY_NAME) {
+            output_file_name = it.second;
+            output_file_.open(output_file_name);
+        }
+    }
+
+    if (!output_file_.is_open()) {
+        throw dds::core::IllegalOperationError(
+                "Open output file: " + output_file_name);
```

Comment:
Error should be "Error opening output file..." or "Failed to open output file..."

Reply:
Done.